python/ray/experimental/data/impl/reader.py

```diff
@@ -16,14 +20,78 @@ def read_file(path: str,
     Returns The contents of the file. If `include_paths` is True, a tuple of
       the path and the contents of the file.
     """
+    parsed = urllib.parse.urlparse(path)
+
     if filesystem:
         if isinstance(filesystem, _S3FileSystemWrapper):
             filesystem = filesystem.unwrap()
         contents = filesystem.open_input_stream(path).readall()
     else:
-        contents = open(path, "rb").read()
+        if parsed.scheme == "s3":
+            contents = download_single_s3_file(parsed.netloc,
+                                               parsed.path.strip("/"))
+        else:
+            contents = open(path, "rb").read()
 
     if include_paths:
-        return path, contents
+        return (path, contents)
```

Comment:
Nit: Parentheses not necessary when returning a tuple. Or are you trying to make it extra explicit that a tuple is being returned?
```suggestion
        return path, contents
```